### ISSUE TITLE
ORT 1.9.0 Build: nvcc error   : 'ptxas' died due to signal 11 (Invalid memory reference)

### ISSUE THREAD

**Issue by danielecazzari** [microsoft/onnxruntime#9222]
**Describe the bug**
I’m trying to compile the RT version 1.9.0 with CUDA, Tensor RT and OpenVINO EP and I’m facing an error with nvcc at the end of the process. 

**Urgency**
I need to be able to build the runtime by this week to be able to embed the new version in the next cadence.

**System information**
   OS: CentOS7 
   NVCC Cuda compilation tools, release 10.2, V10.2.89
   CUDA 10.2.89 (CUDA 10.2 patches have not been applied)
   CuDNN 8.0.2
   TensorRT  7.1.3.4
   NCCL 2.8.4

**To Reproduce**
```
   ONNXRUNTIME_REPO=https://github.com/microsoft/onnxruntime
   ONNXRUNTIME_BRANCH=v1.9.0

   git clone --recursive -b ${ONNXRUNTIME_BRANCH} ${ONNXRUNTIME_REPO}  && \
   cd onnxruntime/cmake/external/onnx-tensorrt && \
   git remote update && \
   git checkout 7.1
```

    ./build.sh --config Release --build_shared_lib --build --update --parallel \
    --skip_tests \
    --skip_submodule_sync  \
    --use_cuda --cuda_home /usr/local/cuda --cudnn_home /usr/lib64 \
    --use_tensorrt --tensorrt_home /usr/lib64 \
    --use_openvino $DEVICE \
    --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER)

**Crash Snippet**
```
[100%] Built target onnxruntime_shared_lib_test
nvcc error   : 'ptxas' died due to signal 11 (Invalid memory reference)
nvcc error   : 'ptxas' core dumped
gmake[2]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/code/onnxruntime/onnxruntime/core/providers/cuda/math/topk_impl.cu.o] Error 139
gmake[1]: *** [CMakeFiles/onnxruntime_providers_cuda.dir/all] Error 2
gmake: *** [all] Error 2
Traceback (most recent call last):
  File "/code/onnxruntime/tools/ci_build/build.py", line 2324, in <module>
    sys.exit(main())
  File "/code/onnxruntime/tools/ci_build/build.py", line 2245, in main
    build_targets(args, cmake_path, build_dir, configs, num_parallel_jobs, args.target)
  File "/code/onnxruntime/tools/ci_build/build.py", line 1174, in build_targets
    run_subprocess(cmd_args, env=env)
  File "/code/onnxruntime/tools/ci_build/build.py", line 625, in run_subprocess
    return run(*args, cwd=cwd, capture_stdout=capture_stdout, shell=shell, env=my_env)
  File "/code/onnxruntime/tools/python/util/run.py", line 42, in run
    completed_process = subprocess.run(
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/cmake', '--build', '/code/onnxruntime/build/Linux/Release', '--config', 'Release', '--', '-j12']' returned non-zero exit status 2.

```
**Full Logs**
[onnx190buildLog.docx](https://github.com/microsoft/onnxruntime/files/7254310/onnx190buildLog.docx)

Regards

Daniele 
SAS Institute

**Comment by jywu-msft** [comment]
is there a specific cuda arch you are targeting? 
can you try adding  --cmake_extra_defines CMAKE_CUDA_ARCHITECTURES=52  or something like that?

**Comment by danielecazzari** [comment]
We do not target any specific architecture, we aim to support whatever architecture that CUDA 10.2 supports. Do you know where to find the list?

**Comment by jywu-msft** [comment]
can you try with just 52 with the option I mentioned above, just to narrow whether there's some issue with compiling for certain architectures? cuda 10.2 is pretty old, any chance you guys will update in the near future?

**Comment by danielecazzari** [comment]
Hi George,
I did several test and I found that 52 works and also all those architecture:
'CMAKE_CUDA_ARCHITECTURES=30;35;37;50;52;60;61;70'

The issue happen again when I add to above cmake command the architecture 75. This should be the Turing architecture and as per my understanding is supported by CUDA 10.2. Do you have any idea why this does not compile or if there are know issue?

About CUDA upgrade I will try to request this again, but as you know it does not depends on me.

Thanks a lot.

Daniele

**Comment by jywu-msft** [comment]
that's good to hear that you are unblocked now.
yes, it's my understanding 75 should be supported with 10.2 (maybe you can try testing with latest patches/updates for cuda 10.2?)
we can ask Nvidia if there are any known issues.

**Comment by danielecazzari** [comment]
Hi George, 
I verified and 75 is the architecture of the quadro rtx 6000 I use to build and test. It is also a used in many other scenario, so supporting this architecture is very important for us.

I was able to build version 1.8.0 of the ONNX Runtime and even the ONNX Master branch till one month ago without having to specify any CUDA architecture so I think this issue might be due to some changes done recently.

Could you check with Nvidia if they have any suggestion for this problem?

**Comment by jywu-msft** [comment]
The build from one month ago didn't include building for 75. We found out that if you did not specify CMAKE_CUDA_ARCHITECTURES explicitly, CMAKE will default it to 52. 
Your build should still work, right? Have you tested the rtx6000? The driver should JIT compile gpu specific sass at runtime and then cache it. 
We will check with Nvidia about the build issue you encountered.

**Comment by jywu-msft** [comment]
additionally, can you test with latest cuda 10.2 patches if you have not already done so?

**Comment by danielecazzari** [comment]
1) Yes the build artifact are working also on rtx6000 even if architecture is not included. Not sure if this add time during loading phase.
2) Unfortunately remove parallel does not fi
[onnx190buildLog-NoParallel.docx](https://github.com/microsoft/onnxruntime/files/7292433/onnx190buildLog-NoParallel.docx)
3) I think that build script use latest 10.2 but feel free to suggest changes
[Dockerfile.centos7_dev3_lvl1_cuda_tensort_openvino.txt](https://github.com/microsoft/onnxruntime/files/7292462/Dockerfile.centos7_dev3_lvl1_cuda_tensort_openvino.txt)
